sources/vega/governance.proto

```diff
@@ -168,11 +168,17 @@ message ProposalTerms {
   // Timestamp (Unix time in seconds) when voting closes for this proposal,
   // constrained by `minClose` and `maxClose` network parameters
   int64 closing_timestamp = 1;
-  // Timestamp (Unix time in seconds) when proposal gets enacted (if passed),
-  // constrained by `minEnact` and `maxEnact` network parameters
-  int64 enactment_timestamp = 2;
   // Validation timestamp (Unix time in seconds)
-  int64 validation_timestamp = 3;
+  int64 validation_timestamp = 2;
+
+  oneof enactment {
+    // Timestamp (Unix time in seconds) when proposal gets enacted (if passed),
+    // constrained by `minEnact` and `maxEnact` network parameters
+    int64 enactment_timestamp = 107;
+    // Last Update is the time duratuon in int64 that was last added to the timestamp field
+    int64 enactment_last_update = 108;
```

Comment:
I'm not sure if this is the best place for this field because it will mean that someone can set it in a `ProposalSubmission` when submitting any new proposal. Whats it for? Something related to checkpoints?

Reply:
This is about validation of new future and update future: https://github.com/vegaprotocol/vega/pull/5668/files#diff-1527092608a66a12e24c11a95fc2ca1e9d2b73063b35c78ad06651b45b37bef8R484,
https://github.com/vegaprotocol/vega/pull/5668/files#diff-1527092608a66a12e24c11a95fc2ca1e9d2b73063b35c78ad06651b45b37bef8R247

It is set only in https://github.com/vegaprotocol/vega/pull/5668/files#diff-92b369672ec5368c6bacbb63f9c3717a133c96f06f473e577ca711abefbcbe23R78
Because when the enactment time is updated, the termination should also be updated, otherwise if the checkpoint happens too much forward in time - we can have termination timestamp before enactment under the hood

Reply:
It should not be here surely. The enactment time cannot be updated in a proposal. We should be able to fix all this just with validation in the core. No changes in the proto are required I believe.

Reply:
All right, thanks!